devdoc/singlylinkedlist2_requirements.md

```diff
@@ -0,0 +1,176 @@
+singlylinkedlist2 requirements
+================
+
+## Overview
+
+SinglyLinkedList is module that provides the functionality of a singly linked list, allowing its user to add, remove and iterate the list elements.
+
+## Exposed API
+
+```c
+typedef struct SLIST_ENTRY_TAG
+{
+    struct SLIST_ENTRY_TAG *next;
+} SLIST_ENTRY, *PSLIST_ENTRY;
+
+typedef bool (*SLIST_MATCH_FUNCTION)(PSLIST_ENTRY listEntry, void* matchContext);
```

Comment:
>listEntry [](http://example.com/codeflow?start=50&length=9)

Since this is a new module can we use the same nameing scheme like in the C standard instead of camelCase?
This would be list_entry. And the next one would be match_context

Reply:
done